src/filter/Filter.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+import pandas as pd
+import abc
+from typing import TYPE_CHECKING, Type
+
+if TYPE_CHECKING:
+    from src.filter.FilterRegistry import FilterRegistry
+    from src.metric.SampleMetricManager import SampleMetricManager
+
+
+class Filter(metaclass=abc.ABCMeta):
+    """
+    Abstract static class outlining the general interface for different types
+    of filters.
+    """
+
+    def __new__(cls):
+        raise RuntimeError(f"{cls.__name__} should not be instantiated.")
+
+    @classmethod
+    def get_bit(cls, filter_registry: Type[FilterRegistry]):
+        """
+        Returns the bit associated with the filter.
```

Comment:
mention in docstring body how the bit is derived. can probably be somewhat copied from `Population.get_filtered_data()`, i.e., "The order of the bits are analogous to the order of filters in the FilterRegistry." (use variable name instead of static class preferably)